### PR TITLE
Taylor feedback on step serializer

### PR DESCRIPTION
**Pull request by styliii**
https://trello.com/c/FiH7wFIU/60-data-format-requests-from-taylor

1. Process info will now be included in step serializer
2. Status information on process changed from TODO => to do. 

image

Still pending:
Should we update resources to be an object? What attributes does it need, other than title and url?

### CODE REVIEW

**Review by keithtom — APPROVED** [review]
I like the step_options pattern.

I feel like maybe we can have default options to include (or any other serializer options) but also support the scenario where an API caller can explicitly add other include options to override the default options.   I think that'd be easy to add after the fact.  But it does make me wonder if we should just put the burden on the API caller...  easy enough to fix later on since we control both sides for now

File: gems/workflow/app/models/workflow/instance/process.rb [hunk 1]
```diff
@@ -23,9 +23,18 @@ def effort
...
+      completed_count = steps.where(completed: true).count
+
+      case completed_count
+      when 0
+        return "to do"
+      when step_count
+        return "done"
+      else
+        return "in progress"
+      end
```
[the remark this answers is not in the document]
Comment (styliii) [reply gems/workflow/app/models/workflow/instance/process.rb]: yup in the next PR

**Comment by styliii** [comment]
right, i'll work on the counter cache next

File: gems/workflow/app/models/workflow/instance/process.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (styliii) [reply gems/workflow/app/models/workflow/instance/process.rb]: > Really minor but can we put these strings as constants? They are pretty special strings. And when we write tests we'll be able to reference them.

did this in a different PR.

**Pull request merged**